File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int xor1 = 0, xor2 = 0;
        int n = nums.size();
        
        // XOR all the numbers in the array
        for(int i = 0; i < n; i++) {
            xor2 = xor2 ^ nums[i];  // XOR of all elements in nums
            xor1 = xor1 ^ i;        // XOR of indices from 0 to n-1
        }
        
        xor1 = xor1 ^ n;  // XOR with n (as n is missing from the indices)
        
        // XOR of xor1 and xor2 gives the missing number
        return xor1 ^ xor2;
    }
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int l = nums.length;
        int res = l*(l+1)/2;
        int sum=0;
        int mis=0;

        for(int i=0;i<l;i++){
            sum += nums[i];
        }

        mis = res - sum;

        return mis;
    }
}